### DISCUSSION TITLE
The filesession.cpp example doesn't work

### DISCUSSION THREAD

**Discussion by yurivict** [daniele77/cli#194] [General]
The executable just exits with exit code 0.

cli-2.0.2
clang-14
FreeBSD 13.1

**Comment by daniele77** [comment]
As you can see from the code:

    std::ifstream infile("input.txt");
    if (!infile)
    {
        std::cerr << "File input.txt not found in current directory!\n";
        return 1;
    }
    std::ofstream outfile("output.txt");
    if (!outfile)
    {
        std::cerr << "Can't write file output.txt in the current directory!\n";
        return 1;
    }
    CliFileSession input(cli, infile, outfile);

This example takes the input from the file `input.txt` and puts its output to the file `output.txt`.
So, the behavior you're observing is right: it must exit with exit code 0 and produce a file named `output.txt` (providing it finds an input file named` input.txt`).

**Comment by yurivict** [comment]
Thank you for the explanation.
